### PR TITLE
fix(snippetgen): fix client streaming samples

### LINKED ISSUE #1014: snippetgen: streaming methods have syntax errors
Generated snippets for streaming methods have incorrect syntax:

https://github.com/googleapis/gapic-generator-python/blob/dbfb45ac6a0dc0b14efc895d7c584062b40ff2c0/tests/integration/goldens/logging/samples/generated_samples/logging_generated_logging_v2_logging_service_v2_tail_log_entries_async.py#L36-L44

### PR DESCRIPTION
**Pull request by busunkim96**
Fixes #1014 and unblocks #1043.

NOTE: Some real world APIs expect the first request to pass a config ([example](https://github.com/googleapis/googleapis/blob/1d163315aecf9d8a3b77eb9e1549a0b353103b32/google/cloud/speech/v1/cloud_speech.proto#L101-L119)) so the generated samples will not work out of the box. This will be addressed when the new sample config language is sorted out.

### CODE REVIEW

File: gapic/samplegen/samplegen.py [hunk 1]
```diff
@@ -551,16 +551,6 @@ def validate_and_transform_request(
...
-        client_streaming_forms = {
-            types.CallingForm.RequestStreamingClient,
-            types.CallingForm.RequestStreamingBidi,
-        }
-
-        if len(base_param_to_attrs) > 1 and calling_form in client_streaming_forms:
-            raise types.InvalidRequestSetup(
-                "Too many base parameters for client side streaming form"
-            )
-
```
Comment (busunkim96) [inline gapic/samplegen/samplegen.py:563]: @software-dov 👋  I'm having trouble understanding this error, would you remember what this was written to prevent?

Some streaming method request messages have 2+ required fields ([pubsub](https://github.com/googleapis/googleapis/blob/bcfd386c224fded5f5c5be7a17bc5884a2a52f59/google/pubsub/v1/pubsub.proto#L1083-L1162)), so the current snippetgen logic adds two `field`s to the dictionary. That causes this error to be raised. Are fields expected to be set differently for streaming?


```py
# Request passed to `validate_and_transform_request`
[
   {
      "field":"subscription",
      "value":"projects/{project}/subscriptions/{subscription}"
   },
   {
      "field":"stream_ack_deadline_seconds",
      "value":2813
   }
]
```

I see two unit tests that raise this error, but I don't see any pre-existing tests showing a valid sample config for a client streaming method.



https://github.com/googleapis/gapic-generator-python/blob/a8d2b3a2c11984d8677e4e6547a0cbb76d9ed915/tests/unit/samplegen/test_samplegen.py#L1353-L1402

Comment (software-dov) [reply gapic/samplegen/samplegen.py:563]: I really don't remember what was going on here. My _guess_ is that I was mildly incorrect about the actual python method signature convention and I was trying to imply that we only wanted a single request to be put into a list...?

TL;DR: It's probably a bug or at least a misreading of the spec on my part.

**Comment by busunkim96** [comment]
This is ready for review, PTAL @software-dov @parthea.

File: tests/integration/goldens/logging/samples/generated_samples/logging_generated_logging_v2_logging_service_v2_tail_log_entries_async.py [hunk 2]
```diff
@@ -38,8 +38,17 @@ async def sample_tail_log_entries():
         resource_names=['resource_names_value_1', 'resource_names_value_2'],
     )
 
+    # This method expects an iterator which contains
+    # 'logging_v2.TailLogEntriesRequest' objects
+    # Here we create a generator that yields a single `request` for
+    # demonstrative purposes.
+    requests = [request]
```
Comment (busunkim96) [inline tests/integration/goldens/logging/samples/generated_samples/logging_generated_logging_v2_logging_service_v2_tail_log_entries_async.py:45]: One alternative is:

```py
requests = iter([requests])
```
I chose the generator over `iter()` to nudge folks towards generating requests on demand.

Opinions/suggestions on the best way to present client streaming welcome!

**Review by parthea — APPROVED** [review]
